spec/httparty/request_spec.rb

```diff
@@ -784,6 +784,260 @@
     end
   end
 
+  describe "a request that 307 redirects" do
```

Comment:
For those that are similar you could just wrap the specs with array iteration and interpolate the code where necessary.

```ruby
[300, 301, 302, 303].each do |code|
  # spec stuff ehre
end
```